From 98ba3dfcc2aa8c3eb00f3f394e33460be8a447c4 Mon Sep 17 00:00:00 2001
From: Alexander Usyskin <alexander.usyskin@intel.com>
Date: Mon, 31 Mar 2014 17:59:24 +0300
Subject: [PATCH] mei: make return values consistent across the driver

Follow-up for bits missed in
commit 7ca96aa278f8b9983184e318b06a0ed9ad0297b8
mei: make return values consistent across the driver

Change-Id: Ic3cac3d0018b1a8b2f068367d05352b13b3e8d12
Signed-off-by: Alexander Usyskin <alexander.usyskin@intel.com>
Signed-off-by: Tomas Winkler <tomas.winkler@intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
(cherry picked from commit b9e08aca6bf0b695ebc23a5e3721ef353af17b7e)

Signed-off-by: Nagaraj S <sivasankaranx.nagaraj@intel.com>
---
 drivers/misc/mei/bus.c    |    2 +-
 drivers/misc/mei/client.c |    5 ++---
 2 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/misc/mei/bus.c b/drivers/misc/mei/bus.c
index aa5ceee..d87813e 100644
--- a/drivers/misc/mei/bus.c
+++ b/drivers/misc/mei/bus.c
@@ -247,7 +247,7 @@ static int ___mei_cl_send(struct mei_cl *cl, u8 *buf, size_t length,
 		return id;
 
 	if (length > dev->me_clients[id].props.max_msg_length)
-		return -EINVAL;
+		return -EFBIG;
 
 	cb = mei_io_cb_init(cl, NULL);
 	if (!cb)
diff --git a/drivers/misc/mei/client.c b/drivers/misc/mei/client.c
index b849258..9e0be66 100644
--- a/drivers/misc/mei/client.c
+++ b/drivers/misc/mei/client.c
@@ -724,10 +724,9 @@ int mei_cl_read_start(struct mei_cl *cl, size_t length)
 
 	cb->fop_type = MEI_FOP_READ;
 	if (mei_hbuf_acquire(dev)) {
-		if (mei_hbm_cl_flow_control_req(dev, cl)) {
-			rets = -ENODEV;
+		rets = mei_hbm_cl_flow_control_req(dev, cl);
+		if (rets < 0)
 			goto out;
-		}
 
 		list_add_tail(&cb->list, &dev->read_list.list);
 	} else {
-- 
1.7.9.5

